src/com/tictacgo/data/Board.java

```diff
@@ -70,18 +79,18 @@ public Board(Player startingPlayer, int height, Context c) {
 		this.height = height;
 
 		/**
-		 * Initializes the pieces ArrayList to all zeros
+		 * Initializes the spaces ArrayList to all empty Spaces
 		 */
-		pieces = new ArrayList<>(SIDE_LENGTH);
-		pieces.add(new ArrayList<ArrayList<Piece>>(SIDE_LENGTH));
-		pieces.add(new ArrayList<ArrayList<Piece>>(SIDE_LENGTH));
-		pieces.add(new ArrayList<ArrayList<Piece>>(SIDE_LENGTH));
-        for (int i = 0; i < SIDE_LENGTH; i++) {
-            // Each space can have up to 2 overlapping pieces.
-            pieces.get(i).add(new ArrayList<Piece>(2));
-            pieces.get(i).add(new ArrayList<Piece>(2));
-            pieces.get(i).add(new ArrayList<Piece>(2));
-        }
+		spaces = new ArrayList<>(SIDE_LENGTH);
+    pieces = new ArrayList<>(SIDE_LENGTH * SIDE_LENGTH * 2);
+
+    for (int i = 0; i < SIDE_LENGTH; i++) {
+      spaces.add(new ArrayList<Space>(SIDE_LENGTH));
+
+      spaces.get(i).add(new Space());
+      spaces.get(i).add(new Space());
+      spaces.get(i).add(new Space());
+    }
```

Comment:
This is another instance where storing the result of `spaces.get(i)` would be a cleaner choice.  Or perhaps another way of writing it:

```java
for(int i = 0; i < SIDE_LENGTH; i++) {
  // Init currentSpace with three Spaces
  List<Space> currentSpace = new ArrayList<>(Arrays.asList(new Space(), new Space(), new Space()));
  spaces.add(currentSpace);
}
```